contracts/external/nexus/WrappedCoveredPrincipalTokenFactory.sol

```diff
@@ -0,0 +1,61 @@
+// SPDX-License-Identifier: Apache-2.0
+pragma solidity 0.8.0;
+
+import { WrappedCoveredPrincipalToken, Authorizable, EnumerableSet } from "./WrappedCoveredPrincipalToken.sol";
+
+/// @author Element Finance
+/// @title WrappedCoveredPrincipalTokenFactory
+contract WrappedCoveredPrincipalTokenFactory is Authorizable {
+    using EnumerableSet for EnumerableSet.AddressSet;
+
+    // Enumerable list of wrapped tokens that get created from the factory.
+    EnumerableSet.AddressSet private _WrappedCoveredPrincipalTokens;
+
+    // Emitted when new wrapped principal token get created.
+    event WrappedCoveredPrincipalTokenCreated(
+        address indexed _baseToken,
+        address indexed _owner
+    );
+
+    /// @notice Initializing the owner of the contract.
+    constructor(address owner_) {
+        _authorize(owner_);
+        setOwner(owner_);
+    }
+
+    /// @notice Allow the owner to create the new wrapped token.
+    /// @param  _baseToken Address of the base token / underlying token that is used to buy the wrapped positions.
+    /// @param  _owner Address of the owner of wrapped futures.
+    /// @return address of wrapped futures token.
+    function create(address _baseToken, address _owner)
+        external
+        onlyOwner
```

Comment:
Why do we need only owner here? In general we don't want to restrict deployments to core team or DAO unless there's a strong reason to do so.